src/plugins/oSnap/components/TransactionBuilder/ContractInteraction.vue

```diff
@@ -40,22 +40,27 @@ const selectedMethod = computed(
 const parameters = ref<string[]>([]);
 
 function updateTransaction() {
-  if (!isValueValid || !isToValid || !isAbiValid) return;
   try {
+    if (!isToValid) {
+      throw new Error('"TO" address invalid');
+    }
+    if (!isAbiValid) {
+      throw new Error('ABI invalid');
+    }
+    if (!isValueValid) {
+      throw new Error('Value invalid');
+    }
+    // throws is method params are invalid
     const transaction = createContractInteractionTransaction({
       to: to.value,
       value: value.value,
       abi: abi.value,
       method: selectedMethod.value,
       parameters: parameters.value
     });
-
-    if (validateTransaction(transaction)) {
-      emit('updateTransaction', transaction);
-      return;
-    }
-  } catch (error) {
-    console.warn('ContractInteraction - Invalid Transaction:',error);
+    emit('updateTransaction', transaction);
+  } catch {
+    props.setTransactionAsInvalid();
```

Comment:
this is essentially what I'm doing in each transaction type.
If validation fails, we throw, then in the catch block we keep the data the same upstream, but we add an invalid flag.
the form doesn't lose it's state. so as soon as the user inputs data that doesn't cause the `create<TransactionType>` to throw, we can update state upstream with the formatted Tx and set it to be valid